client/client_test.go

```diff
@@ -68,23 +70,24 @@ func TestAddLeaf(t *testing.T) {
 		},
 	} {
 		client := New(logID, test.client, testonly.Hasher, env.PublicKey)
-		client.SetMaxTries(1)
-
-		if err, want := client.AddLeaf(ctx, []byte(test.desc)), codes.DeadlineExceeded; grpc.Code(err) != want {
-			t.Errorf("AddLeaf(%v): %v, want, %v", test.desc, err, want)
-			continue
+		{
+			ctx, cancel := context.WithDeadline(context.Background(), time.Now().Add(timeout))
```

Comment:
In this particular case, it won't cause test flakiness because I'm only using the timeout for the parts of the test that I want to timeout.